lms/djangoapps/instructor/views/api.py

```diff
@@ -2700,51 +2703,45 @@ def send_email(request, course_id):
     targets = json.loads(request.POST.get("send_to"))
     subject = request.POST.get("subject")
     message = request.POST.get("message")
+    # optional, this is a date and time in the form of an ISO8601 string
+    schedule = request.POST.get("schedule", "")
+    # optional, this is the timezone captured from the author's browser when requesting a scheduled email
+    browser_timezone = request.POST.get("browser_timezone", "")
+
+    # If this is a scheduled bulk email request then we try to convert the requested schedule date to UTC. We do this
+    # before we attempt to create the email object instance in case there is an issue as we don't want to have an
+    # orphaned email object that will never be sent.
+    if schedule:
+        try:
+            schedule = _convert_schedule_to_utc_from_local(schedule, browser_timezone, request.user)
+            _determine_valid_schedule(schedule)
+        except ValueError as error:
+            error_message = f"Error occurred while attempting to create a scheduled bulk email task: {error}"
+            log.error(f"{error_message}")
+            return HttpResponseBadRequest(repr(error_message))
 
-    # allow two branding points to come from Site Configuration: which CourseEmailTemplate should be used
```

Comment:
There is no change in behavior here. I just moved this functionality to dedicated functions elsewhere in the file. I was trying to improve the readability of the `send_email` function.